src/analysis.zig

```diff
@@ -1446,17 +1446,56 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
+                    switch (node_tags[field_param]) {
+                        .string_literal => {
+                            const field_name = handle.tree.tokenSlice(main_tokens[field_param]);
+                            break :fld_name_blk field_name;
+                        },
+                        .identifier,
+                        .field_access,
+                        => {
+                            var resolved_type = (try analyser.resolveTypeOfNodeInternal(.{
+                                .node = field_param,
+                                .handle = handle,
+                            })) orelse return null;
+                            switch (resolved_type.type.data) {
+                                // const name = "name"
+                                .other => |n| {
+                                    if (resolved_type.handle.tree.nodes.items(.tag)[n] != .string_literal) return null;
+                                    break :fld_name_blk offsets.nodeToSlice(resolved_type.handle.tree, n);
+                                },
+                                // const name: []const u8 = "name";
+                                .pointer => |pointer| {
+                                    if (pointer.elem_ty.type.data != .ip_index) return null;
+                                    var str_node = pointer.elem_ty.type.data.ip_index.node orelse return null;
+                                    str_node += 3; // XXX str_node points to the "u8", + 3 gets "name"
+                                    const elem_ty_tree = pointer.elem_ty.handle.tree;
+                                    if ((str_node > elem_ty_tree.nodes.len - 1) or (elem_ty_tree.nodes.items(.tag)[str_node] != .string_literal)) return null;
+                                    break :fld_name_blk offsets.nodeToSlice(elem_ty_tree, str_node);
+                                },
+                                else => return null,
+                            }
+                        },
+                        else => return null,
+                    }
+                };
 
-                const field_name = handle.tree.tokenSlice(main_tokens[field_param]);
+                // Need at least one char between the quotes, eg "a"
                 if (field_name.len < 2) return null;
 
                 var lhs = (try analyser.resolveTypeOfNodeInternal(.{
                     .node = params[0],
                     .handle = handle,
                 })) orelse return null;
 
-                return try resolveFieldAccess(analyser, lhs, field_name[1 .. field_name.len - 1]);
+                return try resolveFieldAccess(analyser, lhs, field_name[1 .. field_name.len - 1])
+                // Try `@"field_name"`
+                orelse try resolveFieldAccess(
+                    analyser,
+                    lhs,
+                    try std.fmt.allocPrint(analyser.arena.allocator(), "@{s}", .{field_name}),
+                );
```

Comment:
This is a workaround for an issue with DocumentScope. It adds a declaration with the name `@"chip1"` instead of `chip1`. This should be fixed in DocumentScope and not here.
And also, this is a very confusing place to add a comment.

Reply:
Yes, but the point was to make the `orelse` stand out :)

Do let me know if there's a better strat to trim `@"` from `name`.